src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/utils/LockService.java

```diff
@@ -0,0 +1,231 @@
+                .fetchSource(true);
+
+            final UpdateResponse updateResponse = client.update(updateRequest).actionGet();
+
+            return new LockModel(updateLock, updateResponse.getSeqNo(), updateResponse.getPrimaryTerm());
+        } catch (VersionConflictEngineException e) {
+            logger.debug("could not acquire lock {}", e.getMessage());
+        } catch (DocumentMissingException e) {
+            logger.debug("Document is deleted. This happens if the job is already removed and this is the last run." +
+                "{}", e.getMessage());
+        }
+        return null;
+    }
+
+    private LockModel createLock(final LockModel tempLock) throws IOException {
+        try {
+            final IndexRequest request = new IndexRequest(LOCK_INDEX_NAME)
+                .id(tempLock.getLockId())
+                .source(tempLock.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS))
+                .setIfSeqNo(SequenceNumbers.UNASSIGNED_SEQ_NO)
+                .setIfPrimaryTerm(SequenceNumbers.UNASSIGNED_PRIMARY_TERM)
+                .create(true);
+            final IndexResponse indexResponse = client.index(request).actionGet();
+            return new LockModel(tempLock, indexResponse.getSeqNo(), indexResponse.getPrimaryTerm());
+        } catch (VersionConflictEngineException e) {
+            logger.debug("Lock is already created. {}", e.getMessage());
+            return null;
+        }
+    }
+
+    private LockModel findLock(final String lockId) {
+        GetRequest getRequest = new GetRequest(LOCK_INDEX_NAME).id(lockId);
+        GetResponse getResponse = client.get(getRequest).actionGet();
+        if (!getResponse.isExists()) {
+            return null;
+        } else {
+            try {
+                XContentParser parser = XContentType.JSON.xContent()
+                    .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, getResponse.getSourceAsString());
+                parser.nextToken();
+                return LockModel.parse(parser, getResponse.getSeqNo(), getResponse.getPrimaryTerm());
+            } catch (IOException e) {
+                return null;
+            }
+        }
+    }
+
+    public boolean release(final LockModel lock) {
+        try {
+            if (lock == null) {
+                logger.info("Lock is null. Nothing to release.");
+                return false;
+            }
+
+            final LockModel releaseLock = new LockModel(lock, true);
+            final LockModel releasedLock = updateLock(releaseLock);
+            return releasedLock != null;
+        } catch (IOException e) {
+            logger.info("IOException", e);
+            throw new IllegalArgumentException("IOException", e);
+        } catch (Exception e) {
+            logger.info("Exception", e);
+            throw new IllegalArgumentException("Exception", e);
+        }
+    }
+
+    public boolean deleteLock(final String lockId) {
```

Comment:
Is this used anywhere?

Reply:
This is being used `postDelete` in `JoSweeper`